unittests/Core/DepsBuildEngineTest.cpp

```diff
@@ -222,7 +222,7 @@ TEST(DepsBuildEngineTest, BogusConcurrentDepScan) {
     DynamicRule(const KeyType& key, std::vector<std::string>& builtKeys)
       : Rule(key), builtKeys(builtKeys) { }
     Task* createTask(BuildEngine&) override {
-      builtKeys.push_back(key);
+      builtKeys.push_back(key.str());
```

Comment:
Feels like `builtKeys` should store KeyType.

Reply:
Sure. Probably worth to have a little more explicit coverage of keytype.